src/zarr/core/metadata/v3.py

```diff
@@ -135,8 +135,8 @@ def parse_storage_transformers(data: object) -> tuple[dict[str, JSON], ...]:
 
 class V3JsonEncoder(json.JSONEncoder):
     def __init__(self, *args: Any, **kwargs: Any) -> None:
-        self.indent = kwargs.pop("indent", config.get("json_indent"))
         super().__init__(*args, **kwargs)
+        self.indent = config.get("json_indent")
```

Comment:
What do you think about this instead?

```suggestion
        self.indent = kwargs.pop("indent") or config.get("json_indent")
```

alternatively, we could do this:

```python
if self.indent is None:
    self.indent = config.get("json_indent")

Reply:
the [signature](https://docs.python.org/3/library/json.html#json.JSONEncoder) for `JSONEncoder.__init__` isn't so bad, why don't we just use it explicitly?

Reply:
e.g., 
```python
class V3JsonEncoder(json.JSONEncoder):
    def __init__(self, *, skipkeys=False, ensure_ascii=True, check_circular=True, allow_nan=True, sort_keys=False, indent=None, separators=None, default=None) -> None
```

Reply:
Like this? [7a442e1](https://github.com/zarr-developers/zarr-python/pull/2546/commits/7a442e1b700e0d908732e21be89250ce48202560)
Strangely, that also fixes the strange whitespace problem, so the test is now much simpler!